src/lxml/apihelpers.pxi

```diff
@@ -564,11 +564,14 @@ cdef int _setAttributeValue(_Element element, key, value) except -1:
     if not element._doc._parser._for_html:
         _attributeValidOrRaise(tag)
     c_tag = _xcstr(tag)
-    if isinstance(value, QName):
-        value = _resolveQNameText(element, value)
+    if value is None:
```

Comment:
XML serialization produces the empty string as value, even if it is None. Is there a way to tell which tree you are in?